src/config.ts

```diff
@@ -51,53 +52,83 @@ export default class ConfigService {
    */
   #currentVersion: string | null = null;
 
-  constructor(path: string) {
-    this.#uri = path;
+  constructor(uri: string) {
+    this.#uri = uri;
   }
 
-  async resolve() {
-    let url: URL | null = null;
-
-    try {
-      url = new URL(this.#uri);
-    } catch {
-      // Not a valid URL
+  /**
+   * Fetch and return the configuration module.
+   *
+   * @param deadline When a deadline is provided, fetching the module via HTTP
+   *                 will be aborted if the deadline is exceeded. Specified in
+   *                 milliseconds.
+   */
+  async fetchConfig(deadline: number) {
+    if (!this.configIsViaHttp) {
+      return require(/* webpackIgnore: true */ this.#uri).default;
     }
 
-    const isHttpUrl = url?.protocol?.startsWith('http');
-    logger.info(`Resolving render config via ${isHttpUrl ? 'HTTP' : 'provided file'}`);
+    let config: any = null;
 
-    const config = isHttpUrl
-      ? await loadViaHttp(this.#uri)
-      : require(/* webpackIgnore: true */ this.#uri).default;
+    // Handle fetch deadline via an AbortController
+    const ac = new AbortController();
+    const deadlineTimeout = setTimeout(() => ac.abort(), deadline);
 
+    config = await loadViaHttp(this.#uri, ac);
+    clearTimeout(deadlineTimeout);
```

Comment:
Instead of using the abort signal (which feels quite cumbersome to me), why not something like this:
```js
let deadlineTimeout;
const timeoutPromise = new Promise((_, reject) => (deadlineTimeout = setTimeout(() => reject(new Error("Request timed out."))), deadline);
try {
  config = await Promise.race([loadViaHttp(this.#uri), timeoutPromise]);
catch (err) {
  ...
}
clearTimeout(deadlineTimeout);
```

We can also abstract this as:
```js
const promiseWithTimeout = (promise, timeout) => {
  let timer;
  return Promise.race([
    promise.finally(() => clearTimeout(timer)),
    new Promise((_, reject) => (timer = setTimeout(() => reject(new Error("Timed out."))), timeout),
  ]);
};

...

const config = await promiseWithTimeout(loadViaHttp(this.#uri), deadline);
```

Note: it is okay to leave the timeout promise in limbo (never resolved or rejected) as it won't prevent GC from collecting it once nothing references it: https://stackoverflow.com/a/36735167/90297